db.go

```diff
@@ -24,8 +24,10 @@ type DB struct {
 }
 
 // New Creates a new DB instance given a WebSocket URL.
-func New(url string) (*DB, error) {
-	ws, err := websocket.NewWebsocket(url)
+// NewWithOptions Creates a new DB instance given a WebSocket URL and options.
+// Options are passed to the websocket library.
+func New(url string, options ...websocket.Option) (*DB, error) {
```

Comment:
Right I see where the confusion is! I think @plally and I were thinking more along the lines of
```
type SurrealDbOption struct {
   WsOption websocket.Option
}

func New(url string, options ... SurrealDbOption) (*DB, error) {
   wsOptions := make([]websocket.Option, 0)
   for option in options {
      if !IsEmpty(option.WsOption) {
          wsOptions = append(wsOptions, option.WsOption)
      }
   }
   ...
}
```

Reply:
That way we can add other options as and when and handle them as necessary without breaking the signature.